components/brave_wallet/browser/cardano/cardano_provider_impl.cc

```diff
@@ -122,85 +131,126 @@ void CardanoProviderImpl::GetNetworkId(GetNetworkIdCallback callback) {
 
   delegate_->WalletInteractionDetected();
 
-  // Mocked values for development usage.
-  std::move(callback).Run(0, nullptr);
+  std::move(callback).Run(
+      IsCardanoMainnetKeyring(account_id->keyring_id) ? 1 : 0, nullptr);
 }
 
 void CardanoProviderImpl::GetUsedAddresses(GetUsedAddressesCallback callback) {
   auto account_id = GetAllowedSelectedAccount();
   if (!account_id) {
     std::move(callback).Run(
-        {}, mojom::CardanoProviderErrorBundle::New(
-                kAPIErrorRefused, kAccountNotConnectedError, nullptr));
+        std::nullopt,
+        mojom::CardanoProviderErrorBundle::New(
+            kAPIErrorRefused, kAccountNotConnectedError, nullptr));
     return;
   }
 
   delegate_->WalletInteractionDetected();
 
-  // Mocked values for development usage.
-  std::move(callback).Run({"1", "2", "3"}, nullptr);
+  std::vector<std::string> result;
+  for (auto& address :
+       brave_wallet_service_->GetCardanoWalletService()->GetUsedAddresses(
+           account_id)) {
+    auto cardano_address = CardanoAddress::FromString(address->address_string);
+    CHECK(cardano_address);
```

Comment:
it seems odd to crash if the address isn't correct - Can you confirm this is just to sanity check because we expect that the wallet service would never return an invalid address? Would this be true if the account_id is invalid when calling GetUsedAddresses too?

Reply:
wallet service must return a valid address